From ffd2b553fd9e1037d25122c6b749e6f4d1632df0 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 5 Apr 2018 10:28:17 +0200
Subject: [PATCH] v2v: OVF: fix ovf:id for VirtualSystem in OVirt flavour

When writing the OVF in OVirt flavour, write the actual UUID of the
VM as ovf:id attribute for <VirtualSystem>, instead of a dummy value.

Suggested by Arik Hadas in
https://www.redhat.com/archives/libguestfs/2018-April/msg00005.html

(cherry picked from commit 9dce43931a19510be1b6d21ce67d14a4136ce241)
---
 v2v/create_ovf.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v2v/create_ovf.ml b/v2v/create_ovf.ml
index ccde960df..c29b8421c 100644
--- a/v2v/create_ovf.ml
+++ b/v2v/create_ovf.ml
@@ -688,7 +688,7 @@ let rec create_ovf source targets guestcaps inspect
 
       (match ovf_flavour with
       | OVirt ->
-        e "VirtualSystem" ["ovf:id", "out"] !content_subnodes
+        e "VirtualSystem" ["ovf:id", vm_uuid] !content_subnodes
       | RHVExportStorageDomain ->
         e "Content" ["ovf:id", "out"; "xsi:type", "ovf:VirtualSystem_Type"]
           !content_subnodes
-- 
2.17.2

